url74c52881d6_AndrewBourgeois_JGWCAConstants_tgz-pJ8-be_demmel_fun_jgwcaconstants_Tutorial4J8/src/be/demmel/fun/jgwcaconstants/Version.java

```diff
@@ -0,0 +1,67 @@
+package be.demmel.fun.jgwcaconstants;
+
+import java.io.InputStream;
+import java.util.Properties;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class Version {
+
+    private static final Logger logger = LoggerFactory.getLogger(Version.class);
+    // Build the static information when the class is loaded
+    private static final String FULL_VERSION, VERSION;
+    // The path inside the library where the properties file generated by Maven is located
+    //TODO: find a way to fix the location of the effective POM and .properties so that this Version class can be reused in every project
+    private static final String POM_PATH = "/META-INF/maven/be.demmel.fun/jgwcaconstants/pom.properties";
+    // The message that will be printed when any kind of error occurs during the retrieval of the information
+    private static final String ERROR_MESSAGE = "[!ERROR!]";
+    
+    static {
+        String fullVersion = null;
+        
+        StringBuilder sb = new StringBuilder();
+        try {
+            Properties props = loadProperties();
+            sb.append(props.get("artifactId")).append(" ");
+            sb.append(props.get("version"));
+            fullVersion = sb.toString();
+        } catch (Exception ex) {
+            logger.error("An unexpected problem occured loading {}:", POM_PATH,ex);
+            fullVersion = ERROR_MESSAGE;
+        }
+        
+        FULL_VERSION = fullVersion;
+        
+        String version = null;
+        
+        try {
+            Properties props = loadProperties();
+            version = props.get("version").toString();
+        } catch (Exception ex) {
+            logger.error("An unexpected problem occured loading the Maven properties of the module: ", ex);
+            version = ERROR_MESSAGE;
+        }
+        
+        VERSION = version;
+    }
+
+    public static void main(String[] args) {
+        System.out.println(FULL_VERSION);
+    }
+
+    public static String getVersion() {
+        return VERSION;
+    }
+    
+    // Loads the properties generated by Maven in the build process
+    private static Properties loadProperties() throws Exception {
+        InputStream stream = Version.class.getResourceAsStream(POM_PATH);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *stream*. Currently, there are execution paths that do not contain closure statements, for example, when Properties.load throws an exception. Ensure *stream* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).